presto-cli/src/main/java/com/facebook/presto/cli/StatusPrinter.java

```diff
@@ -443,4 +457,92 @@ private static double percentage(double count, double total)
         }
         return min(100, (count * 100.0) / total);
     }
+
+    private class ProgressStats
+    {
+        private final long timestampNanos;
+        private final long processedBytes;
+        private final long processedRows;
+
+        public ProgressStats(long timestampNanos, long processedBytes, long processedRows)
+        {
+            this.timestampNanos = timestampNanos;
+            this.processedBytes = processedBytes;
+            this.processedRows = processedRows;
+        }
+
+        public long getTimestampNanos()
+        {
+            return timestampNanos;
+        }
+
+        public long getProcessedBytes()
+        {
+            return processedBytes;
+        }
+
+        public long getProcessedRows()
+        {
+            return processedRows;
+        }
+    }
+
+    private class ProgressStatsWindow
+    {
+        private final long windowSize;
```

Comment:
Introduced a new class of which using linkedlist to keep the window of progessStats.